### PR TITLE
feat(httplib2): add support for httplib2

### PR DESCRIPTION
**Pull request by ranrib**
In addition, fixed issue with flask wrapper when DISABLE_EPSAGON=TRUE

### CODE REVIEW

File: epsagon/__init__.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def dummy_wrapper(func):
     step_lambda_wrapper = dummy_wrapper  # pylint: disable=C0103
     azure_wrapper = dummy_wrapper  # pylint: disable=C0103
     python_wrapper = dummy_wrapper  # pylint: disable=C0103
-    EpsagonFlask = dummy_wrapper  # pylint: disable=C0103
```
Comment (galbash) [inline epsagon/__init__.py]: is no one using this? you are gonna break someones code (like ours)

Comment (galbash) [inline epsagon/__init__.py]: export with both name, maybe add a deprecation warning

File: epsagon/events/httplib2.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        :param kwargs: wrapt's kwargs
+        :param start_time: Start timestamp (epoch)
+        :param response: response data
+        :param exception: Exception (if happened)
+        """
+
+        super(Httplib2Event, self).__init__(start_time)
+        self.event_id = 'httplib2-{}'.format(str(uuid4()))
+
+        # Params can be set via args or kwargs.
```
Comment (galbash) [inline epsagon/events/httplib2.py]: a better approch would be to define an args unroller:
```
def unroller(uri='N/A', method='N/A', body=None, headers=None):
    return uri, method, body, headers

uri, method, body, headers = unroller(*args, **kwargs)
```

File: epsagon/events/httplib2.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                'request_headers',
+                headers
+            )
+
+        try:
+            if body:
+                add_data_if_needed(
+                    self.resource['metadata'],
+                    'request_body',
+                    simplejson.loads(body)
```
Comment (galbash) [inline epsagon/events/httplib2.py]: import simplejson as json. why are we even using this? it just bloats our package

File: epsagon/events/httplib2.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                )
+        except (TypeError, ValueError):
```
Comment (galbash) [inline epsagon/events/httplib2.py]: There is a JSONDecodeError no?

File: epsagon/events/httplib2.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+        # Extract only json responses
+        try:
+            if response_body:
+                add_data_if_needed(
+                    self.resource['metadata'],
+                    'response_body',
+                    simplejson.loads(response_body)
+                )
+        except (TypeError, ValueError):
```
Comment (galbash) [inline epsagon/events/httplib2.py]: same

File: epsagon/events/httplib2.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    Factory class, generates Httplib2 event.
+    """
+
+    @staticmethod
+    def create_event(wrapped, instance, args, kwargs, start_time, response,
+                     exception):
+        """
+        Create an event according to the given api_name.
+        """
+        url = kwargs.get('uri', args[0] if len(args) >= 1 else None)
```
Comment (galbash) [inline epsagon/events/httplib2.py]: same (unroll)

File: epsagon/__init__.py (hunk 1 above)
Comment (ranrib) [reply epsagon/__init__.py]: The current `EpsagonFlask` breaks the convention actually. Users expects to import `flask_wrapper`

Comment (ranrib) [reply epsagon/__init__.py]: EpsagonFlask is not listed in the `__all__`, So I don't think that anyone is doing `from epsagon import EpsagonFlask`

File: epsagon/events/httplib2.py (hunk 4 above)
Comment (ranrib) [reply epsagon/events/httplib2.py]: JSONDecodeError inherits from ValueError

File: epsagon/events/httplib2.py (hunk 3 above)
Comment (ranrib) [reply epsagon/events/httplib2.py]: Done

File: epsagon/events/httplib2.py (hunk 2 above)
Comment (ranrib) [reply epsagon/events/httplib2.py]: Awesome! Done

File: epsagon/events/httplib2.py (hunk 4 above)
Comment (galbash) [reply epsagon/events/httplib2.py]: so catch the specific error... everything inherits from Object maybe just catch it and thats it

File: epsagon/events/httplib2.py (hunk 5 above)
Comment (galbash) [reply epsagon/events/httplib2.py]: same

**Pull request merged**